### PR TITLE
Dealing with contaminated requests

### PR DESCRIPTION
**Pull request by simonnineteen90**
https://eaflood.atlassian.net/browse/SFI-2094

### CODE REVIEW

File: test/integration/filter-email-addresses.test.js [hunk 1]
```diff
@@ -252,4 +253,39 @@ describe('filter email addresses by event', () => {
 
     expect(result).toStrictEqual([...emailAddresses.flat(), ...splitDevEmailAddresses, ...splitRpaEmailAddresses])
   })
+
+  test('should return a flat array of devEmailAddresses when an event with name of "batch-processing-payment-request-invalid" is given', async () => {
+    const batchProcessingEvent = { ...DEV_EVENT }
+    const result = filterEmailAddresses(batchProcessingEvent)
+
```
Comment (marctemp) [inline test/integration/filter-email-addresses.test.js]: @simonnineteen90 

Suggestion to remove whitespace

File: test/integration/filter-email-addresses.test.js [hunk 2]
```diff
@@ -252,4 +253,39 @@ describe('filter email addresses by event', () => {
 
     expect(result).toStrictEqual([...emailAddresses.flat(), ...splitDevEmailAddresses, ...splitRpaEmailAddresses])
   })
+
+  test('should return a flat array of devEmailAddresses when an event with name of "batch-processing-payment-request-invalid" is given', async () => {
+    const batchProcessingEvent = { ...DEV_EVENT }
```
Comment (marctemp) [inline test/integration/filter-email-addresses.test.js]: @simonnineteen90 

Could be simplified to `const batchProcessingEvent = DEV_EVENT`

File: test/integration/filter-email-addresses.test.js [hunk 3]
```diff
@@ -252,4 +253,39 @@ describe('filter email addresses by event', () => {
...
+    const result = filterEmailAddresses(batchProcessingEvent)
+
+    expect(result).toStrictEqual(splitDevEmailAddresses)
+  })
+
+  test('should return a flat array of devEmailAddresses when an event with name of "responses-processing-quarantine-error" is given', async () => {
+    const batchProcessingEvent = { ...DEV_EVENT, name: 'responses-processing-quarantine-error' }
```
Comment (marctemp) [inline test/integration/filter-email-addresses.test.js]: @simonnineteen90 

Variable name here doesn't make sense as its not a batch processing error but a responses processing error.

Could use the reserved variable `event` here (see line 138 for inspo)

File: test/mock-event.js [hunk 4]
```diff
@@ -86,8 +86,26 @@ const ERROR_EVENT = {
   }
 }
 
+const DEV_EVENT = {
```
Comment (marctemp) [inline test/mock-event.js]: @simonnineteen90 

Do you think we should havva generic `RPA_EVENT`, this subsequent generic `DEV_EVENT` and DEV_AND_RPA_EVENT`?

And when we go to use them for e.g. as you have to check tht dev email addresses are sent off for `name: 'batch-processing-quarantine-error'`, `name: 'batch-processing-error'` we just use the spread operator trick?

What are the pros and cons of doing so?

I.e. this proposed change would:

- delete `BLOCKED_EVENT` and `ERROR_EVENT` from here
- rename `MOCK_RPA_ONLY_EVENT` to `RPA_EVENT`
- require where these were used (and failing now) to be tidied up using the new `DEV_EVENT`, `RPA_EVENT` AND `DEV_AND_RPA_EVENT` w/ our clever spread operator trick used where needs be

File: ffc-pay-alerts/filter/config/dev-events-config.js [hunk 5]
```diff
@@ -2,5 +2,6 @@ module.exports = [
   'batch-processing-quarantine-error',
   'batch-processing-error',
   'payment-request-blocked',
-  'responses-processing-quarantine-error'
+  'responses-processing-quarantine-error',
+  'batch-processing-payment-request-invalid'
```
Comment (johnwatson484) [inline ffc-pay-alerts/filter/config/dev-events-config.js:8]: Should this not be an RPA event?  Siti Agri has sent us something that needs to be corrected.  Or did we agree we'd investigate in the first instance?

**Review by johnwatson484 — COMMENTED** [review]
More of a question than a comment

File: test/mock-event.js (hunk 4 above)
Comment (simonnineteen90) [reply test/mock-event.js]: agree with this so have have implemented as proposed above and amended tests

File: test/integration/filter-email-addresses.test.js (hunk 3 above)
Comment (simonnineteen90) [reply test/integration/filter-email-addresses.test.js]: updated

File: test/integration/filter-email-addresses.test.js (hunk 2 above)
Comment (simonnineteen90) [reply test/integration/filter-email-addresses.test.js]: changed to using `event` as suggested below

File: test/mock-event.js (hunk 4 above)
Comment (marctemp) [reply test/mock-event.js]: Nice good implementing this -- tested and it works like a dream -- much more concise now 👍

File: ffc-pay-alerts/filter/config/dev-events-config.js (hunk 5 above)
Comment (marctemp) [reply ffc-pay-alerts/filter/config/dev-events-config.js:8]: Very good point -- do we need to receive these events or just RPA?

Comment (marctemp) [reply ffc-pay-alerts/filter/config/dev-events-config.js:8]: I think this ties in v well with your point in batch-processor around attaching some contextual information as to 'why' is failed to process -- if we're sending to RPA we'll defo need to include this

Comment (simonnineteen90) [reply ffc-pay-alerts/filter/config/dev-events-config.js:8]: belive this has been resolved in the next PR 
https://github.com/DEFRA/ffc-pay-alerts/pull/15/files

File: test/mock-event.js [hunk 6]
```diff
@@ -86,8 +47,81 @@ const ERROR_EVENT = {
   }
 }
 
+// const BLOCKED_EVENT = {
+//   name: 'payment-request-blocked',
+//   properties: {
+//     id: 'd8899356-6485-48f5-a282-8822f451f77c',
```
Comment (johnwatson484) [inline test/mock-event.js]: Should this be deleted or restored?

**Review by johnwatson484 — APPROVED** [review]

**Pull request merged**